cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/DefaultConsumablePartition.java

```diff
@@ -110,20 +106,22 @@ public void retry() {
 
   @Override
   public void complete() {
-    if (processState != ProcessState.IN_PROGRESS) {
-      throw new IllegalStateException();
-    }
+    assertState(ProcessState.IN_PROGRESS);
     processState = ProcessState.COMPLETED;
   }
 
   @Override
   public void discard() {
-    if (processState != ProcessState.IN_PROGRESS) {
-      throw new IllegalStateException();
-    }
+    assertState(ProcessState.IN_PROGRESS);
     processState = ProcessState.DISCARDED;
   }
 
+  private void assertState(ProcessState expectedState) {
+    if (processState != expectedState) {
+      throw new IllegalStateException(String.format("Partition not in progress: %s", partitionKey));
```

Comment:
I did not notice this earlier: the message would be wrong if expectedState is not IN_PROGRESS